packages/frontend/src/components/AutoComplete.tsx

```diff
@@ -92,7 +94,14 @@ export function AutoComplete<T, U>({
                 className={`p-2 w-full h-full outline-none ${inputClassName}`}
                 type="text"
                 placeholder={placeholder}
-                {...getInputProps()}
+                {...getInputProps({
+                    onKeyDown: (event) => {
+                        if (event.key === "Enter") {
```

Comment:
What happens if we have a professor selected from the auto-complete? Does it navigate to that professors page, or does it just execute the search?

Reply:
It executes the search, would you prefer it to navigate to the selected professor?

Reply:
This should add that feature:

``` typescript
{...getInputProps({
     onKeyDown: (event) => {
          if (event.key === "Enter") {
               event.preventDefault();
               if (highlightedIndex !== null && filteredItems[highlightedIndex]) {
                     window.location.href = `/professor/${filteredItems[highlightedIndex].value}`;
                } else {
                     search(event.currentTarget.value);
                }
          }
     },
})}
```

Reply:
Sorry, I was trying to point out that your change would no longer allow users to directly navigate to the professor page if they have an autocomplete option highlighted. 

The component has already been constructed to properly navigate on both use-cases (an autocomplete option was selected, or the user is trying to query with the search input). We just need to prevent Downshift's default behavior so that it doesn't prevent the parent component from handling the event.

I'd look through the Downshift docs for [customizing combobox behaviors](https://github.com/downshift-js/downshift/tree/master/src/hooks/useCombobox#customizing-handlers) for way to properly prevent Downshift from doing that.

Reply:
Oh, I see, so the code that I displayed above is a workaround because it is preventing the Downshifted default behavior? How can I prevent that from happening?